src/main/kotlin/com/bantolomeus/repository/BookRepository.kt

```diff
@@ -19,20 +19,19 @@ class BookRepository(private val bookFile: String = BOOK_FILE) {
         return books
     }
 
-    fun saveBookIfItNotExists(bookDTO: BookDTO): BookDTO {
-        val book = getBookByName(bookDTO.name)
-        if (book == null) {
+    fun saveBookIfItNotExists(newVersion: BookDTO): BookDTO {
+        val existingVersion = getBookByName(newVersion.name)
+        if (existingVersion == null) {
             val books = getBooks()
-            bookDTO.dateStarted = dateFormat.format(Date())
-            books.add(bookDTO)
-            objectMapper.writeValue(File(bookFile), books)
-            return bookDTO
+            newVersion.dateStarted = dateFormat.format(Date())
+            objectMapper.writeValue(File(bookFile), books + newVersion)
+            return newVersion
         }
 
-        return book
+        return existingVersion
     }
 
-    fun getBooks(): MutableList<BookDTO> {
+    fun getBooks(): List<BookDTO> {
         return try {
             objectMapper.readValue<List<BookDTO>>(File(bookFile)).toMutableList()
```

Comment:
I find it hypocritical to return a mutableList here. The goal of your change was to use List instead of MutableList. And returning a MutableList but saying you return a List just because a MutableList is also a List is more than irritating. If you do not want to change the type to a immutable list, then keep the return value also MutableList.

Reply:
good catch. I will remove the `toMutableList()` calls. It doesn't make much difference though. All the callers can handle immutable lists and the code even looks better ;-) This was what I wanted to prove with my changes.